doc: update doc on tuples (#215)
Substra/substra-documentation: docs/source/documentation/concepts.rst

An algorithm corresponds to an archive (tar or zip file) containing:

>     * One or more Python scripts that implement the algorithm.
>     * A Dockerfile on which the user can specify the required dependencies of the Python scripts.

Reviewer: ```suggestion
* One or more Python scripts that implement the algorithm.
* A Dockerfile on which the user can specify the required dependencies of the Python scripts.
```

Compute plan
""""""""""""
A set of tasks.
Gathering tasks into a single compute plan will lead to a more optimized compute.

Note that you can register a task alone, i.e. not put the task in a compute plan, but Substra will still create a compute plan for you for this specific task.

Task
""""

> The specification of a task. The task can use any inputs (data, algorithms or output from other tasks). The algorithm is expected to write the outputs in files, on paths given as outputs dictionary.

Reviewer: ```suggestion
A task correspond to a computation task. The task can use any inputs (data, algorithms or output from other tasks). The algorithm is expected to write the outputs in files, on paths given as outputs dictionary.
```

Model
^^^^^

> A model or model updates is a potentially large file containing the parameters or updates (gradients) of parameters of a trained model. In the case of a neural network, a model would contain the weights of the neurons. It is either the result of training an algorithm_ with a given dataset_, corresponding to a training task or the result of an aggregate algorithm aggregating models or model updates; corresponding to an aggregation task.

Reviewer: ```suggestion
A model is a potentially large file containing the parameters or updates (gradients) of parameters of a trained model. In the case of a neural network, a model would contain the weights of the neurons. It is either the result of training an algorithm_ with a given dataset_, corresponding to a training task or the result of an aggregate algorithm aggregating models or model updates; corresponding to an aggregation task.
```

Compute plan
""""""""""""

> A set of training, aggregation and testing tasks gathered together towards building a final model.
> Gathering tasks into a single compute plan will lead to a more optimized compute.

Reviewer: ```suggestion
A set of tasks.
Gathering tasks into a single compute plan will lead to a more optimized compute.
```

Note that you can register a task alone, i.e. not put the task in a compute plan, but Substra will still create a compute plan for you for this specific task.

Task
""""

> The specification of a task. The task can use any inputs and algorithm. The algorithm is expected to write the outputs in files, on paths given as outputs dictionary.

Reviewer: ```suggestion
The specification of a task. The task can use any inputs (data, algorithms or output from other tasks). The algorithm is expected to write the outputs in files, on paths given as outputs dictionary.
```

Permissons for a organization to process an asset
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

An organization can execute a task if it has the permission to process all the inputs of the task.
The permission on an asset is defined when creating the asset or when creating the task that will create the asset. Permissions can be defined individually for every organization. Permissions cannot be modified once the asset (or the task producing the asset) is created.

> 

Reviewer: I would remove the paragraph below as it feels redundant:
```

Datasets, algorithms
""""""""""""""""""""
Permissions are defined at creation by their owner for datasets and algorithms.
```
